helm-mt.el

```diff
@@ -73,21 +81,23 @@
         (action . (("Switch to terminal buffer" . (lambda (candidate)
                                                     (switch-to-buffer candidate)))
                    ("Exit marked terminals" 'helm-mt/delete-marked-terms)))))
- 
 
-(defvar helm-mt/term-source-terminal-not-found
-  '((name . "Launch a new terminal")
-    (dummy)
-    (action . (("Launch new terminal" . (lambda (candidate)
-                                          (helm-mt/launch-term candidate)))))))
+(defun helm-mt/term-source-terminal-not-found ()
+  `((name . "Launch a new terminal")
+	(dummy)
+	(action . ,(mapcar (lambda (mode)
+						  `(,(format "Launch new %s" mode) . 
+							(lambda (candidate)
+								(helm-mt/launch-term candidate (quote ,mode)))))
+					  helm-mt/all-terminal-modes))))
 
 ;;;###autoload
 (defun helm-mt ()
   "Custom helm buffer for terminals only."
   (interactive)
   (let ((sources
-        '(helm-mt/term-source-terminals
-          helm-mt/term-source-terminal-not-found)))
+		 `(helm-mt/term-source-terminals
```

Comment:
It was not incorrect before. What changed is that "helm-mt/term-source-not-found" is no longer a constant quoted value. The backquote/comma gets it evaluated so that an action for each term mode can be created based upon the list defined at "helm-mt/all-terminal-modes".